opal/corelib/string/encoding.rb

```diff
@@ -232,13 +232,15 @@ def bytesize(string)
 end
 
 class String
+  attr_reader :bytes
```

Comment:
I'm using this definition to workaround this issue because apparently, `bytes` is iterable/enumerable.

```
Iterable props defined by Opal on core JS objects is empty for strings FAILED
Expected ["0", "1", "2", "bytes"] to equal ["0", "1", "2"]
SpecExpectationNotMetError: Expected ["0", "1", "2", "bytes"] to equal ["0", "1", "2"]
```


But now the linter is unhappy :grimacing: 
Any idea on how to fix this issue (while making the linter happy!)

Reply:
I think for string we need to explicitly define any internal instance variable we use with `Opal.defineProperty` like we do for internal coding: `Opal.defineProperty(String.prototype, 'internal_encoding', #{Encoding::UTF_8})`

Reply:
OK, I will give it a try :+1: